be/src/formats/parquet/file_reader.cpp

```diff
@@ -190,6 +212,34 @@ Status FileReader::_get_footer() {
     return Status::OK();
 }
 
+void FileReader::_build_split_tasks() {
+    if (!_scanner_ctx->enable_split_tasks || _scanner_ctx->split_context != nullptr ||
+        _scanner_ctx->split_tasks == nullptr) {
+        return;
+    }
+
+    size_t row_group_size = _file_metadata->t_metadata().row_groups.size();
+    for (size_t i = 0; i < row_group_size; i++) {
+        const tparquet::RowGroup& row_group = _file_metadata->t_metadata().row_groups[i];
+        bool selected = _select_row_group(row_group);
+        if (!selected) continue;
+        int64_t start_offset = _get_row_group_start_offset(row_group);
+        int64_t end_offset = _get_row_group_end_offset(row_group);
+        auto split_ctx = std::make_unique<SplitContext>();
+        split_ctx->split_start = start_offset;
+        split_ctx->split_end = end_offset;
+        split_ctx->file_metadata = _file_metadata;
+        _scanner_ctx->split_tasks->emplace_back(std::move(split_ctx));
+    }
+    // if only one split task, clear it, no need to do split work.
+    if (_scanner_ctx->split_tasks->size() <= 1) {
+        _scanner_ctx->split_tasks->clear();
```

Comment:
add split_context,  set split_start and split_end?

Reply:
nono, that's not how it works. 

Here we just do split, connector frame work will set right split context for us in next round.